src/lib.rs

```diff
@@ -39,17 +43,17 @@ pub extern "C" fn verify_cose_signature_ffi(
     let cose_signature = unsafe { from_raw(cose_signature, cose_signature_len) };
 
     // Parse the incoming signature.
-    let cose_signatures = decode_signature(cose_signature, &payload);
+    let cose_signatures = decode_signature(&cose_signature, &payload);
     let cose_signatures = match cose_signatures {
-        Ok(signature) => signature,
+        Ok(signatures) => signatures,
         Err(_) => Vec::new(),
     };
-    if cose_signatures.len() < 1 {
+    if cose_signatures.len() == 0 {
         return false;
     }
 
-    let mut result = true;
-    for cose_signature in cose_signatures {
+    // for cose_signature in cose_signatures {
```

Comment:
If this is a comment explaining the following code, let's comment in prose. If it's just commented-out code, let's remove it.

Reply:
This was me being stupid again.